File: 12/httpServer.cs

using System.Net;
using System.Text;

if (!HttpListener.IsSupported)
{
    Console.WriteLine("your os not support HttpClient class");
}
const string ADDR = @"http://127.0.0.1:8001/";
var server = new HttpListener();
//添加监听地址
server.Prefixes.Add(ADDR);
server.Start();
while (true)
{
    var context = server.GetContext();
    var res = context.Response;
    var req = context.Request;
    var client_addr = req.UserHostAddress;
    var buffer = Encoding.UTF8.GetBytes($"\"<html><body><h1>Hello World! {client_addr}</h1></body></html>\"");

    res.ContentLength64 = buffer.Length;
    res.ContentType = "text/html";
    res.OutputStream.Write(buffer, 0, buffer.Length);
    res.Close();
}

File: 12/httpServer.js
const http = require('http')

const server = http.createServer((req, res) => {
  res.writeHead(200, { "Content-Type": "text/plain" })
  res.write("Hello dasdaddasas")
  res.end()
})

server.listen(8000, _ => {
  console.log("server started on port: 8000");
})

